Elaborate more on Local Path Provider (#189)
k3s-io/docs: docs/storage/storage.md

Kubernetes maintainers are actively migrating in-tree volume plugins to CSI drivers. For more information on this migration, please refer [here](https://kubernetes.io/blog/2021/12/10/storage-in-tree-to-csi-migration-status-update/).

## Setting up the Local Storage Provider

> K3s comes with Rancher's Local Path Provisioner and this enables the ability to create persistent volume claims out of the box using local storage on the respective node.

Reviewer: ```suggestion
K3s comes with Rancher's Local Path Provisioner, which provides the ability to use Persistent Volume Claims using local storage on the respective node. Note that because the Persistent Volumes are bound to local paths on the node, pods using these PVs will not be able to reschedule to other nodes, should the node hosting the volume become unavailable.
```

> Below we provide a simple example using a PV and PVC. For more details on the usage of the Local Path Provisioner please reference the official documentation [here](https://github.com/rancher/local-path-provisioner/blob/master/README.md#usage).

Reviewer: ```suggestion
Included below is a sample pod using a PV and PVC. For more details on the usage of the Local Path Provisioner please reference the [official documentation](https://github.com/rancher/local-path-provisioner/blob/master/README.md#usage).
```

> K3s allows for overriding the path used on nodes via `--default-local-storage-path` (see the [Server CLI options](../cli/server.md) for more details including defaults), and further changes to Local Path Provisioner can be made via overriding the provided ConfigMap using [HelmChartConfig](../helm/helm#customizing-packaged-components-with-helmchartconfig). The contents of this ConfigMap is detailed [here](https://github.com/rancher/local-path-provisioner/blob/master/README.md#configuration)

Author: Oh, OK. I'll look at rewording this to explain that you need to use your own deployment. Is there a reason that local-path-provisioner wasn't set up via helm, and is there anything stopping the switch being made?

> K3s allows for overriding the path used on nodes via `--default-local-storage-path` (see the [Server CLI options](../cli/server.md) for more details including defaults), and further changes to Local Path Provisioner can be made via overriding the provided ConfigMap using [HelmChartConfig](../helm/helm#customizing-packaged-components-with-helmchartconfig). The contents of this ConfigMap is detailed [here](https://github.com/rancher/local-path-provisioner/blob/master/README.md#configuration)

Reviewer: Historical reasons mostly. Things that we bundled before adding the embedded helm controller were all managed as flat manifests. When time came to add traefik, it was only available as a helm chart, so we added the helm controller. We added HelmChartConfig support some time later. It's somewhat non-trivial to have helm "take over" existing resources in a nondisruprive way, so we've not prioritized migrating things.
